Issue 86 websocket session confusion

Laukkala opened the pull request:
## Description

Moved the emission of COLLABORATIVE_MODE_STATUS messages so that they are only called when the connected users have actually changed. This should remove the COLLABORATIVE_MODE_STATUS updates being sent to other notebooks.

## Checklists

### Testing

#### General

- [x] I have checked that my test files and functions have meaningful names.
- [x] I have checked that each test tests only a single behavior.
- [x] I have done happy tests.
- [x] I have tested only my own code.
- [x] I have tested at least all public methods.

#### Assertions

- [x] I have checked that my tests use assertions and not runtime overhead.
- [x] I have checked that my tests end in assertions.
- [x] I have checked that there is no comparison statements in assertions.
- [x] I have checked that assertions are in tests and not in helper functions.
- [x] I have checked that assertions for iterables are outside of for loops and both sides of the iteration blocks.
- [x] I have checked that assertions are not tested inside consumers.

#### Testing Data

- [x] I have tested algorithms and anything else with the possibility of unbound growth.
- [x] I have checked that all testing data is local and fully replaceable or reproducible or both.
- [x] I have checked that all test files are standalone.
- [x] I have checked that all test-specific fake objects and classes are in the test directory.
- [x] I have checked that my tests do not contain anything related to customers, infrastructure or users.
- [x] I have checked that my tests do not contain non-generic information.
- [x] I have checked that my tests do not do external requests and are not privately or publicly routable.

#### Statements

- [x] I have checked that my tests do not use throws for exceptions.
- [x] I have checked that my tests do not use try-catch statements.
- [x] I have checked that my tests do not use if-else statements.

#### Java

- [x] I have checked that my tests for Java uses JUnit library.
- [x] I have checked that my tests for Java uses JUnit utilities for parameters.

#### Other

- [x] I have only tested public behavior and not private implementation details.
- [x] I have checked that my tests are not (partially) commented out.
- [x] I have checked that hand-crafted variables in assertions are used accordingly.
- [x] I have tested [Object Equality](https://docs.oracle.com/javase/6/docs/api/java/lang/Object.html#equals%28java.lang.Object%29).
- [x] I have checked that I do not have any manual tests or I have a valid reason for them and I have explained it in the PR description.

### Code Quality

- [x] I have checked that my code follows metrics set in Procedure: Class Metrics.
- [x] I have checked that my code follows metrics set in Procedure: Method Metrics.
- [x] I have checked that my code follows metrics set in Procedure: Object Quality.
- [x] I have checked that my code does not have any NULL values.
- [x] I have checked my code does not contain FIXME or TODO comments.

Laukkala committed:
Replaced removeConnectionFromNote with removeNoteConnection, as the methods are identical.

Laukkala committed:
Added dependency on Junit5

Laukkala committed:
Moved the emission of COLLABORATIVE_MODE_STATUS messages so that they are only called when the connected users have actually changed. This should remove the COLLABORATIVE_MODE_STATUS updates being sent to other notebooks.

Removed references to collaborativeModeList as it's never being meaningfully accessed.

Laukkala committed:
Removed "throws" from tests

Tiihott commented on zeppelin-server/src/main/java/com/teragrep/zep_01/socket/ConnectionManager.java:
```diff
@@ -168,43 +167,28 @@ public void removeConnectionFromAllNote(NotebookSocket socket) {
-        userList.add(noteSocket.getUser());
+      Message message = new Message(Message.OP.COLLABORATIVE_MODE_STATUS);
+      message.put("status", collaborativeStatusNew);
+      if (collaborativeStatusNew) {
```
This if-statement condition is always true.

Laukkala replied:
Removed

Tiihott commented on zeppelin-server/src/test/java/com/teragrep/zep_01/socket/NotebookServerTest.java:
```diff
@@ -683,6 +685,132 @@ public void testNoteRevision() throws IOException {
 
+  @Test
+  public void testCollaborativeModeStatus() {
+    Assertions.assertDoesNotThrow(()->{
```
It would make it easier to debug issues if the assertion only covered the methods that could throw exceptions. Same goes for the next test.

Laukkala replied:
Moved the beginning of the assertion to the latest position where all necessary variables are available

Tiihott commented on zeppelin-server/src/test/java/com/teragrep/zep_01/socket/NotebookServerTest.java:
```diff
@@ -683,6 +685,132 @@ public void testNoteRevision() throws IOException {
+
+      String noteId = "";
+      for (Note note : notebook.getAllNotes()) {
+        if (note.getName().equals(noteName)) {
```
`Assertions.assertEquals(note.getName(), noteName);` would also work so there wouldn't be a need for if-else statement.

Laukkala replied:
Simplified this part to remove the unnecessary loop and if-else statement

Tiihott reviewed (changes requested):
Few issues but otherwise looks OK.

Laukkala committed:
Removed unnecessary if-statement
Moved assertions within tests so that they only cover necessary parts of the test to ease debugging
Simplified the acquisition of NoteIDs

Laukkala committed:
Simplified Assertions to ease debugging

Tiihott reviewed (approved):
Tests pass and changes look good. LGTM.
